src/solver/gemm.cpp

```diff
@@ -87,13 +87,13 @@ bool GemmFwdBase::IsApplicable(const ExecutionContext& ctx,
     const auto& xDesc = problem.GetIn();
     const auto& wDesc = problem.GetWeights();
     const auto& yDesc = problem.GetOut();
-    if(xDesc.GetType() == miopenInt8x4 || xDesc.GetType() == miopenInt8)
-    {
-        // rocBlas needs the output to be int32 always
-        if(yDesc.GetType() != miopenFloat && yDesc.GetType() != miopenInt32 &&
-           yDesc.GetType() != miopenInt8x4)
-            return false;
-    }
+
+    // rocBlas needs the output to be 32-bit always
+    if(xDesc.GetType() == miopenInt8      //
+       && (yDesc.GetType() != miopenFloat //
```

Comment:
checks on lines 93 and 94 are conflicting, i.e. if one is true, other is false and thus the whole check is false. Please review the logic. It could be that these two checks should be in an OR operations.

Reply:
@amberhassaan `(yDesc.GetType() != miopenFloat && yDesc.GetType() != miopenInt32)` means _"the datatype of Y is not FP32 and not INT32"_, i.e. any type except FP32 and INT32. This matches the original code (but can be made better, of course).

Reply:
ok, it was my error in visually parsing. I missed the nested `()`.